segregation/dynamics/divergence_profile.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+import pandas as pd
+
+from scipy.spatial.distance import pdist, squareform
+from scipy.special import rel_entr as relative_entropy
+
+from ..network import compute_travel_cost_matrix
+
+
+def compute_divergence_profiles(
+    gdf, groups, network=None, metric="euclidean"
+):
+    """
+    A segregation metric, using Kullback-Leiber (KL) divergence to quantify the
+    difference in the population characteristics between (1) an area and (2) the total population.
+
+    This function utilises the methodology proposed in
+    Olteanu et al. (2019): 'Segregation through the multiscalar lens'. Which can be
+    found here: https://doi.org/10.1073/pnas.1900192116
+
+    Arguments
+    ----------
+    gdf: geopandas.GeoDataFrame
+        geodataframe with group population counts (not percentages) to be included in the analysis.
+    groups: list
+        list of columns on gdf that contain population counts of interest
+    network: pandana.Network object (optional)
+        A pandana Network object used to compute distance between observations
+    metric : str
```

Comment:
Could we make this work like scikit's `precomputed` argument? So:

1. the "normal" behavior computes distances as is from the geodataframe: `def(gdf, groups, distances=None, metric='euclidean')`
2. The `metric='network'` option treats an input distance matrix as a network:
`def(gdf, groups, distances=network, metric='network')`
3. The `metric='precomputed'` option treats the input distance matrix as given:
`def(gdf, groups, distances=my_matrix, metric='precomputed')`


pdf, groups,

Reply:
I guess what im saying is ultimately the relationship is always represented by some kind of distance matrix, but from the user's perspective I've thought that it's usually preferable to specify the neighborhood using a W or a Network object and have us create the distmat on the backend

I just remembered a `W` is already setup for this(!), so probably the best thing to do is allow *every* index also accept a `metric` argument that [gets pasesed](https://github.com/pysal/segregation/blob/deae25cc7d2a5d61f22fb234bf1bdc8e29b8969d/segregation/_base.py#L233) to the `distance_metric` argument.